src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -349,26 +415,12 @@ export class ExposureNotificationService {
       return;
     }
     captureMessage('past pending summary check');
-    const currentStatus = this.exposureStatus.get();
 
-    if (currentStatus.type === ExposureStatusType.Diagnosed) {
-      const today = getCurrentDate();
-      const cycleEndsAt = new Date(currentStatus.cycleEndsAt);
-      // There is a case where using UTC and device timezone could mess up user experience. See `date-fn.spec.ts`
-      // Let's use device timezone for resetting exposureStatus for now
-      // Ref https://github.com/cds-snc/covid-shield-mobile/issues/676
-      if (daysBetween(today, cycleEndsAt) <= 0) {
-        this.exposureStatus.set({type: ExposureStatusType.Monitoring, lastChecked: currentStatus.lastChecked});
-        return this.finalize();
-      }
-      return this.finalize({needsSubmission: await this.calculateNeedsSubmission()});
-    } else if (currentStatus.type === ExposureStatusType.Exposed) {
-      const today = getCurrentDate();
-      const lastExposureAt = new Date(currentStatus.summary.lastExposureTimestamp || today.getTime());
-      if (daysBetween(lastExposureAt, today) >= EXPOSURE_NOTIFICATION_CYCLE) {
-        this.exposureStatus.set({type: ExposureStatusType.Monitoring, lastChecked: currentStatus.lastChecked});
-        return this.finalize();
-      }
+    const currentExposureStatus = this.exposureStatus.get();
+    const today = getCurrentDate();
+    const updatedExposure = this.updateExposure(currentExposureStatus, today);
+    if (updatedExposure !== currentExposureStatus) {
+      this.exposureStatus.set(updatedExposure);
```

Comment:
I think you should also have a `this.finalize()` call here to match the old code. That updates the `lastChecked` object in the status, and has some extra logic in case `lastChecked` on the current status is missing values.